### PR TITLE
[SCHEMA CHANGE REQUEST] - Add referral status enum and dates for changed status

### PR DESCRIPTION
**Pull request by Danny123234**

### CODE REVIEW

**Review by CosineP — APPROVED** [review]

**Comment by CosineP** [comment]
I assume this change typechecks which I assume means we don't yet touch these fields

**Review by swjeong0825 — CHANGES_REQUESTED** [review]
I think we don't need this enum. If we can use the Dates we have as a status hint, and use the contactedDate or contacted(boolean) variable to indicate that specifically.

File: backend/prisma/schema.prisma [hunk 1]
```diff
@@ -50,18 +58,20 @@ model JobPost {
...
+  employeeId     String
+  candidateId    String
+  jobPostId      String
+  description    String
+  resumeFilePath String?
+  createdDate    DateTime        @default(now())
+  contactedDate  DateTime?
+  finishedDate   DateTime?
+  status         Referral_Status @default(OPEN)
+  deletedDate    DateTime?
```
Comment (slustig210) [inline backend/prisma/schema.prisma:74]: remove deletedDate, just use finishedDate

**Review by slustig210 — CHANGES_REQUESTED** [review]

File: backend/prisma/schema.prisma [hunk 2]
```diff
@@ -4,10 +4,18 @@ generator client {
...
   //url      = "postgresql://postgres:docker@localhost:5432/urefer?schema=public"
 }
 
+enum Referral_Status {
+  OPEN
+  CONTACTED
+  REJECTED
+  HIRED
+  DELETED
+}
```
Comment (slustig210) [inline backend/prisma/schema.prisma:17]: We can add a "CLOSED" enum for when someone else was hired (the JOB POST was closed!) so the rest of the candidates are no longer under consideration

**Review by slustig210 — CHANGES_REQUESTED** [review]

File: backend/prisma/schema.prisma (hunk 2 above)
Comment (slustig210) [reply backend/prisma/schema.prisma:17]: you should not be able to add a referral to a closed/deleted job post
and
you should not be able to edit a referral unless it is open

File: backend/prisma/schema.prisma [hunk 3]
```diff
@@ -41,7 +48,7 @@ model JobPost {
   minYearsExperience Int
   salary             Int
   openings           Int         @default(1)
-  createdDate        DateTime    @db.Timestamp() @default(now())
+  createdDate        DateTime    @default(now()) @db.Timestamp
```
[the remark this answers is not in the document]
Comment (slustig210) [reply backend/prisma/schema.prisma]: i am not sure what db.Timestamp does in the first place. I commented it out

File: backend/prisma/schema.prisma [hunk 4]
```diff
@@ -4,10 +4,20 @@ generator client {
...
 
+enum Referral_Status {
+  SUBMITTED
+  REVIEWED  // the hiring manager looked at the referral and decided not to contact
+  CONTACTED // the referree was contacted, but has not been rejected or hired yet
+  REJECTED  // the referree was contacted, went through the hiring process, and was rejected
+  HIRED     // the referree was contacted, went through the hiring process, and was hired
+  WITHDRAWN // the referrer decided to withdraw the referral while it was SUBMITTED
+  CLOSED    // the job post was closed while the referral was on SUBMITTED
+}
```
Comment (slustig210) [inline backend/prisma/schema.prisma:19]: check these out. Thoughts? Reviewed would use finishedDate as its date.

**Review by slustig210 — APPROVED** [review]

**Pull request closed without merging**